Lib/dis.py

```diff
@@ -427,21 +427,24 @@ def __str__(self):
 class Formatter:
 
     def __init__(self, file=None, lineno_width=0, offset_width=0, label_width=0,
-                       line_offset=0, show_caches=False):
+                       line_offset=0, show_caches=False, *, positions_width=0):
         """Create a Formatter
 
         *file* where to write the output
         *lineno_width* sets the width of the line number field (0 omits it)
         *offset_width* sets the width of the instruction offset field
         *label_width* sets the width of the label field
         *show_caches* is a boolean indicating whether to display cache lines
+        *positions_width* sets the width of the instruction positions field (0 omits it)
 
+        If *positions_width* is specified, *lineno_width* is ignored.
         """
         self.file = file
         self.lineno_width = lineno_width
         self.offset_width = offset_width
         self.label_width = label_width
         self.show_caches = show_caches
+        self.positions_width = positions_width
```

Comment:
It seems unnecessary to have both lineno_width and positions_width, because they will never both be used simultaneously. I think it's tidier to have one field (location_width) for both cases, and a boolean show_positions.

Another option is to pass the location_width along with a function that takes a Positions triplet and returns a string. Then the implementation of the location rendering is outside the Formatter class. Up to you.

Reply:
I considered this option, namely location_width + boolean but I wasn't sure about compatibility. Should I just deprecate the `lineno_width` parameter (e.g., put it to -1 to detect whether it's being used and add `location_width` + boolean)? or can I keep the `lineno_width` name + the boolean? 

Ideally, I wanted to assume that the Formatter class is private and change the signature but I don't want to break people's code (the class has a public name, but is neither documented nor exported).

Reply:
I've changed the interface here: https://github.com/python/cpython/pull/123168/commits/112fe26b52e917a3e03ca6346523595170553cba

Reply:
The Formatter class is new in 3.13. Technically we shouldn't change APIs at this stage, but I think since it's not documented and 3.13 was not released yet, it's safe to assume nobody is using it yet (except me in codoscope).

That said, lineno_width + boolean also works.

Reply:
Err... I went for the `location_width` + boolean + deprecation path. Let me revert it to the `lineno_width` + boolean path. That way, I don't have this deprecation to handle.